sched/signal/sig_dispatch.c

```diff
@@ -760,7 +760,7 @@ int nxsig_dispatch(pid_t pid, FAR siginfo_t *info, bool thread)
            * with the current thread.
            */
 
-          if (stcb != NULL && group == this_task()->group)
+          if (stcb != NULL && group == stcb->group)
```

Comment:
This whole check group == stcb->group is just silly. If stcb != NULL, it is always true....